### PR TITLE
eth2util: consolidate network info

### PR DESCRIPTION
**Pull request by xenowits**
Consolidate `chainID`, `network name` and `fork version` information for supported networks.

category: misc
ticket: none

### CODE REVIEW

**Comment by dB2510** [comment]
@xenowits refactor these too:
https://github.com/ObolNetwork/charon/blob/9486b6d065f65ce4a1d98c61e21d248f39cc3e14/cmd/createcluster.go#L448
https://github.com/ObolNetwork/charon/blob/9486b6d065f65ce4a1d98c61e21d248f39cc3e14/cmd/createdkg.go#L49

**Comment by xenowits** [comment]
Also, in https://github.com/ObolNetwork/charon/blob/9486b6d065f65ce4a1d98c61e21d248f39cc3e14/eth2util/deposit/deposit.go#L202

File: eth2util/network.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+)
+
+// Network contains information about an Ethereum network.
+type Network struct {
+	// ChainID represents chainID of the network.
+	ChainID int64
+	// Name represents name of the network.
+	Name string
+	// ForkVersion represents fork version of the network in hex.
+	ForkVersion string
```
Comment (corverroos) [inline eth2util/network.go:33]: Suggest `ForkVersionHex`

File: eth2util/network.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		ForkVersion: "0x90000069",
+	}
+	Ropsten = Network{
+		ChainID:     3,
+		Name:        "ropsten",
+		ForkVersion: "0x80000069",
+	}
+)
+
+var supportedNetworks = map[Network]bool{
```
Comment (corverroos) [inline eth2util/network.go:64]: rename `allNetworks` and make it a slice

File: eth2util/network.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+// ForkVersionToChainID returns the chainID corresponding to the provided fork version.
+func ForkVersionToChainID(forkVersion []byte) (int64, error) {
+	for network := range supportedNetworks {
+		if fmt.Sprintf("%#x", forkVersion) == network.ForkVersion {
+			return network.ChainID, nil
+		}
+	}
+
+	return -1, errors.New("invalid fork version")
```
Comment (corverroos) [inline eth2util/network.go:80]: Always return zero values on error

File: eth2util/network.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+	// ChainID represents chainID of the network.
+	ChainID int64
+	// Name represents name of the network.
+	Name string
+	// ForkVersion represents fork version of the network in hex.
+	ForkVersion string
+}
+
+var (
+	Mainnet = Network{
```
Comment (corverroos) [inline eth2util/network.go:37]: maybe prefix names with `Network`?

**Review by corverroos — APPROVED** [review]

File: eth2util/network.go (hunk 4 above)
Comment (xenowits) [reply eth2util/network.go:39]: Not sure, will make it more verbose i guess ?

**Pull request merged**